### DISCUSSION TITLE
How to pass secret token to async Server Side Component from Client Side Component (like when using firebase)

### DISCUSSION THREAD

**Discussion by KeisukeNagakawa** [vercel/next.js#42471] [Help]
Hi,  I am thinking of applying firebase js sdk and firebase admin sdk with Next.js 13.  To do that, first I set token like:  
```typescript
'use client';
import { ReactNode } from 'react';
export function GetToken({ children }: { children: ReactNode }) {
  const token = 'token';
  return (
    <div>
 ....
    </div>
  );
}
```

but  I dont know how to save this token to other part of components for Server Side .Could someone help me solve this problem?   Thanks!

**Comment by icyJoseph** [comment]
Say for example that you made a request to an API route, from which you set it as a cookie, then you could do the traditional approach of reading from cookies. There's even a Server Components utility function called, `cookies`, https://beta.nextjs.org/docs/api-reference/cookies - I think that's the most vanilla way to do this. 

Get the token, post it to an API route which might even sign it or obfuscate it in some way and set it as a httpOnly cookie for your PATH = /, then in any other further request the server will see this cookie.

**Reply by KeisukeNagakawa** [reply]
Great!!Thank you so much!

**Reply by thanhtutzaw** [reply]
Why do you need token in firebase sdk ? Are you making firebase auth ? What do you want from firebase and firebase admin ?

**Reply by thanhtutzaw** [reply]
I also learning with next-firebase-auth and they store userinfo in cookies . So we combine cookies + ssr data fetching . It will be good in UX.
